src/generic/triangle_mesh.h

```diff
@@ -836,98 +816,87 @@ namespace oomph
       // It's in the set of elements to be split
       else
       {
-        // [zdec] debug
-        oomph_info << "Splitting element " << e << " at " << el_pt
-                   << " into: " << std::endl;
-        // Vector to get the pointers to the new elements
-        Vector<FiniteElement*> new_el_pt(3, 0);
-        Vector<Node*> new_nod_pt;
-
-        // Split the element
-        split_element_through_centroid<ELEMENT>(
-          el_pt, new_el_pt, new_nod_pt, time_stepper_pt);
-
-        // Add the new elements to new the Vector of pointers
-        new_or_retained_el_pt.push_back(new_el_pt[0]);
-        new_or_retained_el_pt.push_back(new_el_pt[1]);
-        new_or_retained_el_pt.push_back(new_el_pt[2]);
-
-        // Add the nodes to the new Vector of pointers
-        unsigned n_new_nod = new_nod_pt.size();
-        for (unsigned i_new_nod = 0; i_new_nod < n_new_nod; i_new_nod++)
+       // Vector to get the pointers to the new elements
+       Vector<FiniteElement*> new_el_pt(3, 0);
+       Vector<Node*> new_nod_pt;
+       
+       // Split the element
+       split_element_through_centroid<ELEMENT>(
+        el_pt, new_el_pt, new_nod_pt);
+       
+       // Add the new elements to new the Vector of pointers
+       // and doc
+       for (unsigned i=0;i<3;i++)
         {
-          new_or_retained_nod_pt.insert(new_nod_pt[i_new_nod]);
+         new_or_retained_el_pt.push_back(new_el_pt[i]);
+         if (new_elements_stream.is_open())
+          {
+           unsigned nplot=5;
+           new_el_pt[i]->output(new_elements_stream,nplot);
```

Comment:
I'm not opposed to a tecplot style output for doccing the split, but I was careful before to document the old and new nodes as well as the elements. Is this not something users might want to keep an eye on?

Reply:
Already documented (elsehwere, possibly)